apps/evm/lib/evm/logger.ex

```diff
@@ -39,42 +43,42 @@ defmodule EVM.Logger do
       |> Enum.reverse()
       |> Stream.with_index()
       |> Enum.each(fn {value, i} ->
-        value_string =
-          if value == 0,
-            do: "0",
-            else:
-              value
-              |> :binary.encode_unsigned()
-              |> Base.encode16(case: :lower)
-              |> String.trim_leading("0")
-
-        Logger.debug(fn -> "       | #{i}: 0x#{value_string}" end)
+        value_string = stack_value_string(value)
+        Logger.debug(fn -> "       | #{i}: #{value_string}" end)
       end)
     end
   end
 
+  defp current_step(machine_state) do
+    machine_state.step + 1
```

Comment:
Nitpick: adding 1 here feels a bit odd. Can we move the call to `log_opcode_and_gas_left` after the step is incremented? Or maybe initialize step to 1?

Reply:
Love the nitpick because it felt very weird. I'll change this.